### PR TITLE
[RPD-255] improve docstrings in core module

### PR DESCRIPTION
**Pull request by KirsoppJ**
This is a very small PR which extends the `destroy()`, `get()` and `provision()` docstrings slightly, such that the API docs are more easily followed.

## Checklist
Please ensure you have done the following:
* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [x] I have updated the documentation if required.
* [ ] I have added tests which cover my changes.
## Type of change
Tick all those that apply:
Docs improvement
* [ ] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Other (add details above)

### CODE REVIEW

File: src/matcha_ml/core/core.py [hunk 1]
```diff
@@ -91,7 +102,11 @@ def get(
 
 @track(event_name=AnalyticsEvent.DESTROY)
 def destroy() -> None:
-    """Destroys provisioned resources in the cloud.
+    """Destroy the provisioned cloud resources.
+
+    Decommission the cloud infrastructure built by matcha when provision has been called either historically or during
```
Comment (Christopher-Norman) [inline src/matcha_ml/core/core.py:107]: Nit: uppercase `Matcha`

File: src/matcha_ml/core/core.py [hunk 2]
```diff
@@ -120,7 +135,7 @@ def analytics_opt_in() -> None:
 
 
 def remove_state_lock() -> None:
-    """Unlock remote state."""
+    """Unlock the remote state."""
```
Comment (Christopher-Norman) [inline src/matcha_ml/core/core.py:138]: Can we add a sentence about when/why to use this and maybe add a warning about possibly breaking the state file if it is used?

File: src/matcha_ml/core/core.py [hunk 3]
```diff
@@ -132,7 +147,17 @@ def provision(
     password: str,
     verbose: Optional[bool] = False,
 ) -> MatchaState:
-    """Provision cloud resources using templates.
+    """Provision cloud resources using existing matcha templates.
```
Comment (Christopher-Norman) [inline src/matcha_ml/core/core.py:150]: Would `Provision cloud resources using existing Matcha Terraform templates.` work better?

**Review by Christopher-Norman — CHANGES_REQUESTED** [review]
A couple of nits and comments, looking good.

**Review by Christopher-Norman — APPROVED** [review]
LGTM 👍

File: src/matcha_ml/core/core.py [hunk 4]
```diff
@@ -132,7 +152,17 @@ def provision(
     password: str,
```
Comment (JonoCX) [inline src/matcha_ml/core/core.py:152]: The docstring for this wants updating to something more generic, e.g., "the password for the deployment server" (or similar) - users are probably going to think "what's the ZenServer"

File: src/matcha_ml/core/core.py [hunk 5]
```diff
@@ -132,7 +152,17 @@ def provision(
...
     verbose: Optional[bool] = False,
 ) -> MatchaState:
-    """Provision cloud resources using templates.
+    """Provision cloud resources using existing Matcha Terraform templates.
+
+    Provision cloud resources in the location provided. Provide a prefix for the Azure group's  name and a password for
```
Comment (JonoCX) [inline src/matcha_ml/core/core.py:157]: Spacing between "group's" and "name"

**Review by JonoCX — CHANGES_REQUESTED** [review]
Looking good - see comments

I wonder if there's something that could be added to the analytics functions?

File: src/matcha_ml/core/core.py (hunk 5 above)
Comment (KirsoppJ) [reply src/matcha_ml/core/core.py:157]: nice catch

**Review by JonoCX — APPROVED** [review]
LGTM - nice work

**Pull request merged**